emperor/support_files/js/color-view-controller.js

```diff
@@ -193,71 +204,154 @@ define([
       throw new Error("Could not find " + map + " as a colormap.");
     }
 
-    // 1 color and continuous coloring should return the first element of the
-    // map
-    if (numColors === 0 && discrete === false) {
+    // 1 color and continuous coloring should return the first element in map
+    if (values.length == 1 && discrete === false) {
       colors[values[0]] = chroma.brewer[map][0];
       return [colors, gradientSVG];
     }
 
-    if (discrete === false) {
+    //Call helper function to create the required colormap type
+    if (discrete) {
+      colors = ColorViewController.getDiscreteColors(values, map);
+    } else if (scaled) {
+      var info = ColorViewController.getScaledColors(values, map);
+      colors = info[0];
+      gradientSVG = info[1];
+    } else {
+      colors = ColorViewController.getInterpolatedColors(values, map)
+    }
+    return [colors, gradientSVG];
+  };
+
+  /**
+   *
+   * Retrieve a discrete color.
+   *
+   * @param {values} list of objects to generate a color for, usually a
+   * category in a given metadata column.
+   * @param {map} string, name of the discrete color map to use.
+   *
+   * This function will generate a list of coloring values depending on the
+   * coloring scheme that the system is currently using (discrete or
+   * continuous).
+   *
+   * Defaults to use qiime discrete colors if there's no map passed in.
+   *
+   */
+  ColorViewController.getDiscreteColors = function(values, map) {
+    var map = map || 'discrete-coloring-qiime';
+
+    if (map == 'discrete-coloring-qiime') {
+      map = ColorViewController._qiimeDiscrete;
+    } else {
       map = chroma.brewer[map];
-      interpolator = chroma.bezier([map[0], map[3], map[4], map[5], map[8]]);
-      if (scaled === true) {
-        // Assuming we have numeric since scaled called
-        var numericValues = _.map(values, Number);
-        min = _.min(numericValues);
-        max = _.max(numericValues);
-        interpolator = chroma.scale(map).domain([min, max]);
-      }
     }
+    var size = map.length;
+    var colors = {};
+    for (var i = 0; i < values.length; i++) {
+        mapIndex = i - (Math.floor(i / size) * size);
+        colors[values[i]] = map[mapIndex];
+    }
+    return colors;
+  };
 
-    for (var index in values) {
-      if (discrete) {
-        // get the next available color
-        colors[values[index]] = ColorViewController.getDiscreteColor(index, map);
-      }
-      else if (scaled === true) {
-        colors[values[index]] = interpolator(Number(values[index])).hex();
+  /**
+   *
+   * Retrieve a scaled color set with scale bar.
+   *
+   * @param {values} list of objects to generate a color for, usually a
+   * category in a given metadata column.
+   * @param {map} string, name of the discrete color map to use.
+   *
+   * This function will generate a list of coloring values depending on the
+   * coloring scheme that the system is currently using (discrete or
```

Comment:
This block is also out of date, can you remove it? Also, can you add a `@return` section?